Menu/GameShell/10_Settings/Time/timezone_lib_list_page.py

```diff
@@ -0,0 +1,334 @@
+                'file_path': os.path.join(path, f),
+                'is_file': os.path.isfile(os.path.join(path, f))
+            } 
+            for f in os.listdir(path) 
+            ]
+        return elements
+
+    def SyncList(self,path):
+        
+        alist = self.buildDirectoryList(path)
+        if not alist:
+            print("buildDirectoryList empty")
+            return 
+
+        self._MyList = []
+        self._SwapMyList = []
+        
+        start_x  = 0
+        start_y  = 0
+        hasparent = 0
+        if self._MyStack.Length() > 0:
+            hasparent = 1
+            li = ListItem()
+            li._Parent = self
+            li._PosX   = start_x
+            li._PosY   = start_y
+            li._Width  = Width
+            li._Fonts["normal"] = self._ListFont
+            li._MyType = ICON_TYPES["DIR"]
+            li.Init("[..]")
+            self._MyList.append(li)
+        
+        for i,v in enumerate(sorted(alist)):
+            li = ListItem()
+            li._Parent = self
+            li._PosX   = start_x
+            li._PosY   = start_y + (i+hasparent)*ListItem._Height
+            li._Width  = Width
+            li._Fonts["normal"] = self._ListFont
+            li._MyType  = ICON_TYPES["FILE"]
+            
+            if not v['is_file']:
+                li._MyType = ICON_TYPES["DIR"]
+            else:
+                li._MyType  = ICON_TYPES["FILE"]
+            
+            li.Init( v['name']  )
+            li._Path = v["file_path"]
+            
+            
+            self._MyList.append(li)
+
+        for i in self._MyList:
+            self._SwapMyList.append(i)
+        
+    def Init(self):
+        self._PosX = self._Index * self._Screen._Width
+        self._Width = self._Screen._Width
+        self._Height = self._Screen._Height
+
+        self._CanvasHWND = self._Screen._CanvasHWND
+
+        ps = ListPageSelector()
+        ps._Parent = self
+        self._Ps = ps
+        self._PsIndex = 0
+
+        self.SyncList("/usr/share/zoneinfo/posix")
+
+        icon_for_list = MultiIconItem()
+        icon_for_list._ImgSurf = MyIconPool._Icons["sys"]
+        icon_for_list._MyType = ICON_TYPES["STAT"]
+        icon_for_list._Parent = self
+        
+        icon_for_list.Adjust(0,0,18,18,0)        
+        self._Icons["sys"] = icon_for_list
+
+
+        self._BGpng = IconItem()
+        self._BGpng._ImgSurf = MyIconPool._Icons["empty"]
+        self._BGpng._MyType = ICON_TYPES["STAT"]
+        self._BGpng._Parent = self
+        self._BGpng.AddLabel("No timezones found on system!", fonts["varela22"])
+        self._BGpng.SetLableColor(pygame.Color(204,204,204))
+        self._BGpng.Adjust(0,0,self._BGwidth,self._BGheight,0)
+
+
+        self._Scroller = ListScroller()
+        self._Scroller._Parent = self
+        self._Scroller._PosX = self._Width - 10
+        self._Scroller._PosY = 2
+        self._Scroller.Init()
+        
+
+    def ScrollUp(self,Step=1):
```

Comment:
What are  your thoughts on having a common "scroll list" class in the sys.py/UI?
I can see this being handy and I wouldn't want to copy/paste this logic every time for a list view.

Reply:
why is your Menu/GameShell/10_Settings/Time/list_item.py  a binary file?

Reply:
do you mean sys.py/UI/scroller.py?
It's a common UI element used by

>  ./Menu/GameShell/Music Player/play_list_page.py
> ./Menu/GameShell/Music Player/music_lib_list_page.py
> ./Menu/GameShell/10_Settings/ButtonsLayout/__init__.py
> ./Menu/GameShell/10_Settings/Airplane/__init__.py
> ./sys.py/UI/Emulator/rom_list_page.py

for scroll content
maybe Dont need to copy and pase everytime ,
just 
import it